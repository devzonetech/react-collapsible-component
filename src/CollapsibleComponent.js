import React from 'react'
import '!style-loader!css-loader!./css/styles.css'

class CollapsibleComponent extends React.Component {
    
    constructor() {
        super()
    }

    componentDidMount() {
        var acc = document.getElementsByClassName('accordion-head')
        var i

        for (i = 0; i < acc.length; i++) {
            acc[i].addEventListener('click', function() {
                /* Highlight the button - add/remove "active-accordion" class */
                this.classList.toggle('active-accordion')

                /* Hide/show the active panel */
                var panel = this.nextElementSibling
                if (panel.style.display === 'block') {
                    panel.style.display = 'none'
                } else {
                    panel.style.display = 'block'
                }
            })
        }
    }
    
    render() {
        return (<div>
            {this.props.children}
        </div>
        )
    }
}

CollapsibleComponent.propTypes = {
}

CollapsibleComponent.defaultProps = {
}

export default CollapsibleComponent