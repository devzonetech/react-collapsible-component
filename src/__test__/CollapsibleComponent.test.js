import React from 'react'
import renderer from 'react-test-renderer'
import CollapsibleComponent from '../CollapsibleComponent'

describe('CollapsibleComponent', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<CollapsibleComponent />)
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})