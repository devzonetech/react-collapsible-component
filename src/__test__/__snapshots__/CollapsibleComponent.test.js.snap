// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleComponent snapshot renders 1`] = `<div />`;
